codegen/templates/http_client.tmpl

```diff
@@ -206,7 +229,24 @@ func (c *{{$clientName}}) {{$methodName}}(
 
 	{{if $sidecarRouter -}}
 	headers[c.callerHeader] = c.callerName
-	headers[c.calleeHeader] = c.calleeName
+
+	// Set the service name if dynamic routing header is present
+	for headerKey, headerVal := range headers {
+		if routeMap, ok := c.alternateRoutingMap[headerKey]; ok {
```

Comment:
it might be better if you iterate over alternateRoutingMap and see if any of those keys are present in the incoming headers. By doing it this way, even when no alternative routes are configured, you would have iterated over all the headers and did a lookup in the map (which is the common case) and the alternative route headers are quite small in number (usually 1) whereas there are several req headers (10+)